recipes/llvm-core/all/CMakeLists.txt

```diff
@@ -4,11 +4,20 @@ project(conanllvm)
 include(${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup(KEEP_RPATHS)
 
-list(PREPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-if(UNIX)
-  # LLVM 11.1.0 requires the zlib lib dir to be in the library path.
-  # This is not needed for later versions.
-  set(ENV{LIBRARY_PATH} "${CONAN_LIB_DIRS_ZLIB}:$ENV{LIBRARY_PATH}")
+if(LLVM_ENABLE_ZLIB)
+  find_package(ZLIB QUIET)
+
+  # The FindZLIB find module provided by CMake behaves differently than the FindZLIB.cmake script generated by CONAN.
+  # So the following two lines fill in the needed properties.
+  list(GET ZLIB_LIBRARIES 0 ZLIB_LIBRARY)
+  set_property(TARGET ZLIB::ZLIB PROPERTY LOCATION "${ZLIB_LIBRARY}")
+
+  if(UNIX)
+    # Additionally LLVM 11.1.0 requires the zlib lib dir to be in the library path.
+    # This is not needed for later versions and can be removed once
+	# LLVM-12 becomes the oldest supported version.
+    set(ENV{LIBRARY_PATH} "${CONAN_LIB_DIRS_ZLIB}:$ENV{LIBRARY_PATH}")
```

Comment:
~This doesnt work in 2.0~ but this have been migrated

Reply:
I'm not exactly sure what you mean. Is there something I should change?

Reply:
Sorry, I was burning through the review list and went too fast on that comment !


The `CONAN_LIB_DIRS` variables are from the legacy generators that will be removed in 2.0 so this will stop working 

There's no need to change anything, the recipe hasn't been migrated so it's totally expected